### PR TITLE
staffdocs: Add getinvolved page

### PR DESCRIPTION
**Pull request by dkess**
Thanks to everyone who helped prepare the first draft for this! This PR serves as a final round of feedback on this document. This page will be read by a lot of newbies (we expect a big influx this Monday, and I'll mention it in the meeting) so I'd like a lot of eyes on it.

### CODE REVIEW

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
+﻿[[!meta title="Getting Involved"]]
+
+## How can I get involved in the OCF?
+
+Glad you asked! Students from all backgrounds are welcome to join staff. We
+don’t interview or have a selective process for joining-- you simply have to
```
Comment (cg505) [inline ocfweb/docs/docs/staff/getinvolved.md]: eM dAsH pLz

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+We have a curated list of projects at [ocf.io/projects][projects]. Projects
+labeled “newbie” are ones we think are best fit for new staffers, but we are
+happy to provide context and mentorship on any project. In general, don’t worry
+about fully understanding the project description. If you find a something that
+interests you, respond to the thread or reach out to a veteran staffer to get
+more information.
+
+#### GitHub Issues
+
+All of our code is on GitHub (github.com/ocf). Sometimes you can find ways to
```
Comment (cg505) [inline ocfweb/docs/docs/staff/getinvolved.md]: we should make github.com/ocf into a hyperlink

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+     ([printer.rules.yaml][prometheus-printer])
+   * Send a desktop notification if a print jobs fails ([PR #321][puppet-321])
+* [ocfweb][ocfweb] ([issues][ocfweb/issues]) - the source code for our website,
+  www.ocf.berkeley.edu. Written in Python using the Django framework.
+   * Hosts our documentation for users and staffers (including
+     [this page][getinvolved-src]!) ([docs][docs-src])
+   * JSON APIs for lab hours ([api/hours.py][api-hours])
+   * [Account creation][account-register], [password resets][account-chpass],
+     [others][account]
+   * Lab usage statistics ([stats/][ocfweb])
```
Comment (cg505) [inline ocfweb/docs/docs/staff/getinvolved.md]: think you mean `[ocfweb-stats]`

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+   * Form for requesting a virtual host ([lab_reservations/][lab_reservations])
```
Comment (cg505) [inline ocfweb/docs/docs/staff/getinvolved.md]: wat

I think you forgot to update the text, also you probably want `[ocfweb-lab_reservations]`

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+hosting and helping users debug or fix their websites. If you’re interested in
+seeing how this process works, ask a staffer to add you to the RT mailing list.
+Since this is high-volume, we recommend filtering these to a different folder
+so you don’t get notified all the time.
+
+Joining this list doesn’t obligate you to respond to help requests. It’s
+perfectly acceptable to only read tickets as a way to learn more about what we
+do.
+
+If you’re on staff, you can view the ticket archive at rt.ocf.berkeley.edu.
```
Comment (cg505) [inline ocfweb/docs/docs/staff/getinvolved.md]: rt.ocf.berkeley.edu should be a link

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+
+Yes! You can join at any time. If you’re worried about what you’ve missed, we
+recommend reading archives from our mailing list ([ocf.io/announce][announce]).
+These will usually contain links to presentations slides and information about
+what was covered.
+
+**What if I can’t physically be present at weekly meetings?**
+
+You can still be part of OCF staff as the suggestions on this page are useful
+even if you can’t make meetings.
```
Comment (cg505) [inline ocfweb/docs/docs/staff/getinvolved.md]: Feel like currently not quite enthusiastic enough. I'd reword to something like:
> You can still be a part of OCF staff! Even if you can't make meetings, the suggestions on this page are useful for getting started.

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+  network with our Slack organization, so staff members can exercise choice in
+  what chat program they’d like to use. Written in Python.
+* [ircbot][ircbot] ([issues][ircbot/issues]) - a fun chatbot, also used to
+  approve new account requests. Written in Python.
+   * See https://ircbot.ocf.berkeley.edu/ for a listing of IRC bot functionality
+
+#### TODOs
+
+If you’re feeling particularly adventurous, you can [search for the string
+“TODO” across our entire codebase][sourcegraph-todo].
```
Comment (cg505) [inline ocfweb/docs/docs/staff/getinvolved.md]: WAIT NO WHERE DID THE "TODO: add another sentence here?" GO?

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+   * Talk about your interests so we can help you find stuff to work on
+   * Get one-on-one mentoring so you can learn about the vast complex world of
+     OCF infrastructure
+   * Get context and more information on a particular area you’re interested
+     in contributing in
+
+### Slack or IRC
+
+OCF staff often communicate on Slack or IRC. For instructions on how to join
+Slack, see [ocf.io/slack][slack]. Some important channels:
```
Comment (abizer) [inline ocfweb/docs/docs/staff/getinvolved.md]: perhaps note that these are bridged together, and maybe link IRC to thelounge?

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 9]
```diff
@@ -0,0 +1,227 @@
...
+
+These channels are typically very active, and you shouldn’t feel the need to
+read every line to stay updated. You also aren’t required to understand all the
+technical jargon that’s used in #rebuild. Feel free to introduce yourself in
+any of these channels if you’re new, and don’t be afraid to ask questions!
+
+### Hang out in the lab
+
+OCF staff members usually like to socialize in the lab. We’re a friendly bunch,
+so feel free to talk to us! You can also come into the lab after-hours-- just
```
Comment (abizer) [inline ocfweb/docs/docs/staff/getinvolved.md]: emdash

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 10]
```diff
@@ -0,0 +1,227 @@
...
+say you’re interested in joining staff and we’ll let you in.
+
+## FAQ
+
+**I wasn’t able to make the first/second/nth meeting of the semester, can I
+still join?**
+
+Yes! You can join at any time. If you’re worried about what you’ve missed, we
```
Comment (abizer) [inline ocfweb/docs/docs/staff/getinvolved.md]: > worried about what you've missed
perhaps also mention they should feel free/comfortable to ask veteran staff to bring them up to speed

File: ocfweb/docs/docs/staff/getinvolved.md (hunk 6 above)
Comment (abizer) [reply ocfweb/docs/docs/staff/getinvolved.md]: and perhaps mention that staying in touch via IRC or email is absolutely a way to stay 'in the community' even if they can't be physically present

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 11]
```diff
@@ -0,0 +1,227 @@
...
+### In-person meetings
+
+We hold two meetings every Monday:
+
+#### Board of Directors (BoD)
+
+The OCF Board of Directors has open meetings from 7-8 pm every Monday. BoD is
+where we discuss and vote on administrative decisions. If you are interested in
+gaining insight into the internal operations the OCF from an administrative
+perspective, please come to BoD meetings! Meetings always are open to all.
```
Comment (fydai) [inline ocfweb/docs/docs/staff/getinvolved.md]: I think that there should be more description on what happens at BoD meetings, to attract people potentially more into the administrative side of the OCF. I think there should be a bit longer elaborating on the third sentence, perhaps "If you are somebody who wants to figure out how an organization such as the OCF gets and spends its money, what issues the OCF needs to deal with, and all the administrative work that needs to be done to keep the OCF running, please come to BoD meetings!"

I think it would be good to make a list of administrative things that the OCF needs help with to prospective staff, much like the list of technical issues below. I don't believe myself qualified enough to make such a list, but perhaps such a list might include the following.

Finances -
Revenue - include a link to the Fall 2018 buysheet and link some spreadsheet summarizing how our money is being used, and possible ways we may want to spend money on (desktop refresh, more servers, HPC, more stuff to lend)
Income - Briefly summarize (or copy from some previous slides) our funding sources, talk about referendum, etc. Talk about how the OCF needs people to come up with new ways to spend money, and new ways to get money.

Collaborations -
List past collaborations and invite reader to help develop our relationships with various other campus groups and tech companies (Facebook, Cloudless, plus whatever collaborations we've done with campus groups). Insert something like "We need people to look into other collaborations."

Lab maintenance -
We need people to help write and decide on a banning policy, CLA, and figure out how to provide more services that the majority can use other than just printing (perhaps lending stuff?). We need people to talk with <insert long list of roles that keep the OCF running>.

**Comment by cg505** [comment]
Keep in mind that we should get this merged by first meeting tomorrow.

File: ocfweb/docs/docs/staff/getinvolved.md (hunk 7 above)
Comment (dkess) [reply ocfweb/docs/docs/staff/getinvolved.md]: rip wasn't sure if people would get the joke

File: ocfweb/docs/docs/staff/getinvolved.md (hunk 8 above)
Comment (dkess) [reply ocfweb/docs/docs/staff/getinvolved.md]: I'll just link to the docs

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 12]
```diff
@@ -0,0 +1,236 @@
...
+#### TODOs
+
+If you’re feeling particularly adventurous, you can [search for the string
+“TODO” across our entire codebase][sourcegraph-todo].
+
+#### Help requests
+
+During the semester, our [[help@ocf.berkeley.edu mailing list|doc contact]]
+gets around 14 emails per week from the UC Berkeley community. These emails get
+responded to by volunteers like you. Most requests are about requesting virtual
```
Comment (keur) [inline ocfweb/docs/docs/staff/getinvolved.md]: Sentence is awkwardly phrased. How about: "Volunteers like you respond to these emails"

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 13]
```diff
@@ -0,0 +1,236 @@
...
+As a volunteer organization, the OCF’s direction is driven by the interests of
+our members. If you have a new idea for the OCF, we want to help you build it.
+You can use our chat channels or talk to a veteran staffer in person to get a
+starting point and roadmap.
+
+### Staff Hours
+
+During the week, veteran staffers host “staff hours”
+([ocf.io/staffhours][staffhours]), where we provide support to users of our
+services. As a newbie, attending staff hours can serve multiple purposes:
```
Comment (kpengboy) [inline ocfweb/docs/docs/staff/getinvolved.md]:
```suggestion
services. Newbies are encouraged to attend staff hours too.
For a newbie, attending staff hours can serve multiple purposes:
```

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 14]
```diff
@@ -0,0 +1,236 @@
...
+issue.” Here’s a list of our most active repos, with some examples of things we
+can do with them:
+
+* [puppet][puppet] ([issues][puppet/issues]) - contains Puppet modules, which
+  controls the software and configuration on all of our servers.
+   * Install software packages on the desktops
+     ([ocf_desktop/manifests/packages.pp][puppet-desktop-packages])
+   * Configure Firefox to show the "are you sure you want to quit" dialog
+     ([PR #373][puppet-373])
+   * Allow operations staff to refund pages ([PR #368][puppet-368])
```
Comment (kpengboy) [inline ocfweb/docs/docs/staff/getinvolved.md]: It's unlikely that newcomers to the OCF would understand why opstaff wouldn't otherwise be able to refund pages, or why this code belongs in the Puppet repo. I would find a different example.

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 15]
```diff
@@ -0,0 +1,236 @@
...
+   * Monitor our mirrors to see if they’re out of date
+     ([healthcheck script][mirror-healthcheck],
+     [configuring server to call the healtcheck
+     script][mirror-healthcheck-puppet],
+     [Prometheus rules for alerting][prometheus-mirror])
+   * Monitor printers to see if they’re out of paper, toner, or jammed
+     ([printer.rules.yaml][prometheus-printer])
+   * Send a desktop notification if a print jobs fails ([PR #321][puppet-321])
```
Comment (kpengboy) [inline ocfweb/docs/docs/staff/getinvolved.md]: If I were a newcomer, reading through these three examples, I might think that Puppet's role is to monitor services or send notifications to desktops. These examples can be at the least reworded to emphasize what Puppet's role in enabling each of these is.

(For that matter, newcomers probably haven't heard of Prometheus fwiw)

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 16]
```diff
@@ -0,0 +1,236 @@
...
+
+#### GitHub Issues
+
+All of our code is on GitHub ([github.com/ocf][ocf-github]). Sometimes you can find ways to
+contribute by reading open issues, but they can sometimes be hard to understand
+since they’re short on context. Some issues are also marked as “good first
+issue.” Here’s a list of our most active repos, with some examples of things we
+can do with them:
+
+* [puppet][puppet] ([issues][puppet/issues]) - contains Puppet modules, which
```
Comment (kpengboy) [inline ocfweb/docs/docs/staff/getinvolved.md]: Puppet is a tricky repo for devising examples because, in addition to actual configuration code, the Puppet repo also serves as a kind of software repository for various scripts which are only installed on a few hosts. If it were me, I would focus on the theme of configuration management, limiting examples to Puppet code and configuration files (as opposed to, for instance, trying to showcase the variety of files and their functions in this repository).

Some additional examples I think could be worth highlighting, that newcomers would be most likely to understand the significance of:
 - The Postfix file that configures our mail aliases
 - Our DHCP server config
 - Puppet code configuring Apache on our web server

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 17]
```diff
@@ -0,0 +1,236 @@
+﻿[[!meta title="Getting Involved"]]
+
+## How can I get involved in the OCF?
+
+Glad you asked! Students from all backgrounds are welcome to join staff. We
+don’t interview or have a selective process for joining— you simply have to
+express interest and show up to our weekly meetings. If you want to learn more
```
[the remark this answers is not in the document]
Comment (dkess) [reply ocfweb/docs/docs/staff/getinvolved.md]: What do you think the expected time commitment is? I prefer to make it vague since we it varies a lot between staffers

File: ocfweb/docs/docs/staff/getinvolved.md (hunk 15 above)
Comment (dkess) [reply ocfweb/docs/docs/staff/getinvolved.md]: changed "Prometheus" to "Monitoring"

File: ocfweb/docs/docs/staff/getinvolved.md (hunk 16 above)
Comment (dkess) [reply ocfweb/docs/docs/staff/getinvolved.md]: These examples look good, I'll add them. Though I still think the current examples are fine too, as long as there's a big variety I don't think people will get the wrong idea of what Puppet is about.

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 18]
```diff
@@ -0,0 +1,256 @@
...
+#### Board of Directors (BoD)
+
+**Mondays 7-8pm**
+
+BoD is where we discuss and vote on administrative decisions. If you are
+someboday who wants to learn about how an organization such as the OCF gets and
+spends its money, what issues the OCF needs to deal with, and all the
+administrative work that needs to be done to keep the OCF running, please come
+to BoD meetings! Meetings are always open to all. Some examples of what we
+discuss at BoD:
```
Comment (cg505) [inline ocfweb/docs/docs/staff/getinvolved.md]: Most of these things happen outside of BoD, we just discuss progress at BoD (and sometimes vote on things). I kind of think we shouldn't group these examples under the "in-person meetings" category. However, I think this is definitely well-written and good for now. We can think about restructuring this part later.

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 19]
```diff
@@ -0,0 +1,256 @@
...
+
+##### Collaborations
+
+* Industry - we have mutually beneficial relationships with companies like Yelp,
+  Facebook, Kloudless, and more.
+* Campus groups - we want to improve in this area. Bring your ideas to BoD!
+
+##### Lab Administration
+
+We draft policies and discuss new services to provide the UC Berkeley community.
```
Comment (cg505) [inline ocfweb/docs/docs/staff/getinvolved.md]: imo this should be a bullet for consistency even though there's only one item
```suggestion
* We draft policies and discuss new services to provide the UC Berkeley community.
```

File: ocfweb/docs/docs/staff/getinvolved.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (cg505) [reply ocfweb/docs/docs/staff/getinvolved.md]: >...you simply have to express interest and show up to our weekly meetings. **We don't have any time commitment expectations; some staff work many hours each week, and others only have time for occasional contributions.** If you want to learn more...

File: ocfweb/docs/docs/staff/getinvolved.md (hunk 18 above)
Comment (dkess) [reply ocfweb/docs/docs/staff/getinvolved.md]: ahhh this is a draft, wasn't meant to go in this commit, see the gdocs list in #administrivia for comments on this

File: ocfweb/docs/docs/staff/getinvolved.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (dkess) [reply ocfweb/docs/docs/staff/getinvolved.md]: sgtm, added a sentence that conveys this

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 20]
```diff
@@ -0,0 +1,249 @@
+﻿[[!meta title="Getting Involved"]]
+
+## How can I get involved in the OCF?
+
+Glad you asked! Students from all backgrounds are welcome to join staff. We
+don’t interview or have a selective process for joining—you simply have to
```
Comment (ethanhs) [inline ocfweb/docs/docs/staff/getinvolved.md]:
```suggestion
don’t interview or have a selection process for joining—you simply have to
```

I think it is better to say we don't select rather than the process isn't selective (because that implies that there may be selection).

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 21]
```diff
@@ -0,0 +1,249 @@
...
+express interest and show up to our weekly meetings. We don't impose a specific
+time commitment; some volunteers work many hours per week, others contribute
+occasionally. If you want to learn more and be an active participant, there are
+multiple ways to get started:
+
+### Stay up-to-date
+
+We use a mailing list ([ocf.io/announce][announce]) to announce weekly meetings
+and recap past meetings. If you’re joining in the middle of the semester or
+missed a meeting, you can also read the mailing list archives to see what we
```
Comment (ethanhs) [inline ocfweb/docs/docs/staff/getinvolved.md]: It'd be good to link to the archives here.

Comment (dkess) [reply ocfweb/docs/docs/staff/getinvolved.md]: ocf.io/announce is the archives

File: ocfweb/docs/docs/staff/getinvolved.md [hunk 22]
```diff
@@ -0,0 +1,249 @@
...
+and you can unsubscribe at any time.
+
+[[Read more about other mailing lists available to join|doc
+staff/mailing-lists]]
+
+### In-person meetings
+
+We hold two meetings every Monday:
+
+#### Board of Directors (BoD)
```
Comment (kpengboy) [inline ocfweb/docs/docs/staff/getinvolved.md]: Put staff meeting before BoD, since it's more geared toward newcomers? (Or is this a deliberate move to encourage BoD participation?)

Comment (dkess) [reply ocfweb/docs/docs/staff/getinvolved.md]: I can see both sides, but I think it's best to put BoD first because it is held in time before staff meeting. It's less confusing this way.

Comment (kpengboy) [reply ocfweb/docs/docs/staff/getinvolved.md]: On the other hand, though, in order of importance to newcomers, staff meeting would come before BoD...
